### ISSUE TITLE
"Could not load type" warning

### ISSUE THREAD

**Issue by TaffarelJr** [meziantou/Meziantou.Analyzer#218]
**Context**
Rule identifier: _MA0101? (maybe, not sure)_
Target Framework: _.NET Standard 2.0_
Language version: _7.3_

**Describe the bug**
When building my .NETStandard 2.0 class library in VS 16.8.2, I'm getting a warning on each build:

| Field | Value |
|---|---|
| Severity: | Warning |
| Code: | CS8032 |
| Description: | An instance of analyzer Meziantou.Analyzer.Rules.StringShouldNotContainsNonDeterministicEndOfLineAnalyzer cannot be created from C:\Users\\{username}\\.nuget\packages\meziantou.analyzer\1.0.646\analyzers\dotnet\cs\Meziantou.Analyzer.dll: Could not load type 'Meziantou.Analyzer.Rules.StringShouldNotContainsNonDeterministicEndOfLineAnalyzer' from assembly 'Meziantou.Analyzer, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.. |
| Project | {project name} |
| File | |
| Line | 1 |
| Suppression State | Active |

**To Reproduce**
I've not been able to reproduce this warning outside my solution. I've created a brand-new VS solution (with a new .NET Standard 2.0 Class Library project and empty `Class1`), added some verbatim strings to it, and added a reference to `Meziantou.Analyzer`, but the warning does NOT appear in that scenario.

I've tried `git clean -dfx` and rebuild on my VS solution, but that made no difference - the warnings still appear.

I've also tried different versions of `Meziantou.Analyzers`, going back to 1.0.640, but I get the same warnings.

**Additional context**
My VS solution contains additional projects that reference this root class library, including some xUnit test projects that use multi-targeting to test my library against .NET Core 2.1 and .NET Core 3.1. I've been using these analyzers in this solution for several months now, successfully. (Really like them, great work!) Currently I'm trying to add .NET 5.0 as an additional target in my xUnit test projects, and this is when the warnings appeared. (I'm not changing the target of my class library, it's still .NET Standard 2.0; only the test projects get a new target.)

The same warning is actually repeated for each target platform of each xUnit test project as well - so I'm getting 14 instances of the warning in total.

Not sure if it's relevant, but as part of adding .NET 5.0 support to my solution, I've removed the reference to `Microsoft.CodeAnalysis.FxCopAnalyzers` from ALL of my projects, and substituted the new properties instead:
```
    <PropertyGroup>
        <EnableNETAnalyzers>true</EnableNETAnalyzers>
        <AnalysisMode>AllEnabledByDefault</AnalysisMode>
    </PropertyGroup>
```
(I _think_ I'm doing that right; all my projects _should_ be using the new built-in NetAnalyzers in addition to the Meziantou ones; but I don't know how to tell.)

I _assume_ the rule in question is MA0101 - and while I do have 4 verbatim strings in my solution, none of them use explicit line endings (they just wrap to the next line) and none of them are even in the project in question. They're in a project that _references_ my root class library.

I don't know enough about the Analyzer architecture and how it works; I assume it's some kind of configuration issue on my end, but I don't know where else to look. Can you help me track down what might be the cause of this warning?

**Comment by meziantou** [comment]
Hi,

This error can happen when you use an older version of Roslyn.

- Does the project build with `dotnet build`? Are there warnings about analyzers?
- What's the output of `dotnet --info`?
- What's the full version of Visual Studio? (available in Help / About Microsoft Visual Studio)
- What is the version of Roslyn? You can add `#error version` in a cs file of your project. This should report the version of Roslyn in the error window or in the build output.

**Comment by TaffarelJr** [comment]
- `dotnet build` completes successfully with 0 warnings and 0 errors.

- `dotnet --info` gives:
  ```
  .NET SDK (reflecting any global.json):
   Version:   5.0.100
   Commit:    5044b93829

  Runtime Environment:
   OS Name:     Windows
   OS Version:  10.0.18363
   OS Platform: Windows
   RID:         win10-x64
   Base Path:   C:\Program Files\dotnet\sdk\5.0.100\

  Host (useful for support):
    Version: 5.0.0
    Commit:  cf258a14b7

  .NET SDKs installed:
    1.1.14 [C:\Program Files\dotnet\sdk]
    2.2.207 [C:\Program Files\dotnet\sdk]
    3.1.403 [C:\Program Files\dotnet\sdk]
    5.0.100 [C:\Program Files\dotnet\sdk]

  .NET runtimes installed:
    Microsoft.AspNetCore.All 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
    Microsoft.AspNetCore.All 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
    Microsoft.AspNetCore.App 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
    Microsoft.AspNetCore.App 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
    Microsoft.AspNetCore.App 3.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
    Microsoft.AspNetCore.App 5.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
    Microsoft.NETCore.App 1.0.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
    Microsoft.NETCore.App 1.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
    Microsoft.NETCore.App 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
    Microsoft.NETCore.App 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
    Microsoft.NETCore.App 3.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
    Microsoft.NETCore.App 5.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
    Microsoft.WindowsDesktop.App 3.1.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 5.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

  To install additional .NET runtimes or SDKs:
    https://aka.ms/dotnet-download
  ``` 

- Full version of VS is: Microsoft Visual Studio Professional 2019 Version 16.8.2. Do you want the versions of any specific installed products?

- `#error version` gives: `Compiler version: '3.8.0-5.20567.16 (53c5d7d3)'. Language version: 7.3.`

**Comment by TaffarelJr** [comment]
I notice I need to bump up to VS 16.8.4. I'm going to try that to see if it helps.

**Comment by TaffarelJr** [comment]
Well, that seems to have done it. VS 16.8.4, warnings gone. Wish I had noticed that update earlier. 🤦 Thanks so much for your time!
